### ISSUE TITLE
Latest Iris v1.21.4 is loading and crashing with Minecraft Fabric 1.21.3

### ISSUE THREAD

**Issue by shemarmc** [IrisShaders/Iris#2575]
### Minecraft Version

1.21.3

### Iris Version

iris 1.8.4+mc1.21.4

### Sodium Version

sodium-fabric-0.6.5-snapshot+mc1.21.3-build.492.jar

### Operating System

Linux

### What is your GPU?

AMD RX7600

### Java Version

Java 16

### Reproduction Steps

1. Place iris-fabric-1.8.4+mc1.21.4.jar in mods folder or install using installer
2. Try to launch Fabric Loader 1.21.3
3. Game Crashes

[2024-12-21-4.log](https://github.com/user-attachments/files/18219626/2024-12-21-4.log)

### Crash Report file and latest.log

```shell

```

### Additional context

There is no need for the separate crash report. My folder contains mods for both Fabric Loader 1.21.3 and 1.21.4 including manually installations of Iris 1.8.1 and 1.8.4. I have traced the issue to branch 1.21.4 file `fabric.mod.json` line 41 where version `1.21.x` is written I have changed it to `1.21.4` and recompiled the jar. In this case Fabric will no longer attempt to load `iris 1.8.4+mc1.21.4`  but will instead load `iris 1.8.1+mc1.21.3` as it should. Lastly it should be noted  `iris 1.8.1+mc1.21.3` is also loading in Fabric 1.21.4 and crashing.

[mc1.21.3 iris 1.8.4.log](https://github.com/user-attachments/files/18219626/2024-12-21-4.log)
[mc1.21.3 iris 1.8.1.log](https://github.com/user-attachments/files/18219642/latest.log)
[mc1.21.4 iris 1.8.1.log](https://github.com/user-attachments/files/18219648/latest.log)

**Comment by IMS212** [comment]
...yes, this is right. 1.21.4 is not compatible with 1.21.3 mods, and the other way as well.

**Comment by shemarmc** [comment]
Hi Ims I don't mean to be rude but I don't think you understand the issue. Fabric loader has certain rules that it checks for mod compatibility and loading. In this case that it the `minecraft` version property of the mod that is causing an issue. If the mod is incompatible it should simply be marked incompatible so that it does not load on an incompatible minecraft version. All the mods that I have installed use the minecraft version property appropriately and accurately marking their compatibility. This includes sodium, which is a mod that you also contribute to.

In conclusion the issue is that the mod compatibility is marked incorrectly. Which causes it to be loaded in an incompatible version. I apologize if my explanation was not clear before. I would appreciate an explanation if you would still further choose to not fix this issue.
